packages/ketcher-react/src/Editor.tsx

```diff
@@ -64,10 +64,7 @@ function Editor(props: EditorProps) {
       }
     });
     return () => {
-      // setTimeout is used to disable the warn msg from react "Attempted to synchronously unmount a root while React was already rendering"
-      setTimeout(() => {
-        appRoot.unmount();
-      });
+      appRoot.unmount();
     };
```

Comment:
After some deeper investigation , I found this info from the official site:
`createRoot() controls the contents of the container node you pass in.
Any existing DOM elements inside are replaced when render is called.
Later calls use React’s DOM diffing algorithm for efficient updates.`

My understanding is that DOM elements inside the appRoot would only be updated in the second render in strict mode,  while listeners will be attached twice which causes the memory leak.

While for now, I am not sure the best fix for this issue.

Reply:
Hi Stella! Thank you for the reply.

Can you clarify what you mean by 
>while listeners will be attached twice which causes the memory leak.
I'm not very familiar with the code base, but wouldn't the listeners be removed when we unmount?

If the listeners aren't being destroyed and are attached twice, then this would only occur when we are in strict mode. i.e., dev mode. I think this is okay?

>While for now, I am not sure the best fix for this issue.

Would there be an issue with just removing the setTimeout? I'm seeing the error we're getting documented here https://github.com/facebook/react/issues/25675
"Attempted to synchronously unmount a root while React was already rendering".

Reply:
Hi! 
By this `while listeners will be attached twice which causes the memory leak.`, I meant the reason for the original bug I fixed here: https://github.com/epam/ketcher/issues/2912.
Removing the setTimeout seems working here, just I am sure if it is the best practice. 
If we really follow this fix, we may also need to add some extra checks for this function, otherwise the first appRoot will be null after unmounted and there is another error for it.
<img width="291" alt="image" src="https://github.com/epam/ketcher/assets/35161544/d8bf88fc-e413-42a1-856a-a7c34fb5d6b0">
![image](https://github.com/epam/ketcher/assets/35161544/2c900023-9627-452d-8975-2ea34bacedbe)

Reply:
Gotcha thanks! I will look into that. Can I ask how you replicated
>otherwise the first appRoot will be null after unmounted and there is another error for it.

Does this occur in strictmode only or in regular behavior? I'll see what I can do!

Reply:
Yes, the error occurs in strictmode only. If it's not too much trouble, consider checking out this branch, running `npm i` in the root path, and then launching the project to try.

Reply:
Okay I tried seeing what we can do, and I don't think theres a good solution taking account of strictMode.

We can't directly check for truthy in for `_internalRoot`. In general, we really shouldn't mess with the internal fibers. Future react updates would break this.

If we try to defer unmount, we get 
<img width="1200" alt="image" src="https://github.com/epam/ketcher/assets/9488760/666380b2-7ebf-45af-8f07-87c97ee5d22f">

I thought of this, but it feels really hacky. I haven't tried it yet though.
https://dev.to/ag-grid/react-18-avoiding-use-effect-getting-called-twice-4i9e

What are your thoughts on this? In return for working in strict mode, we'll get this error along with the error you tried to silence.

Reply:
I wonder if it is possible to not do this in a useEffect?

Reply:
Hi, as for this solution: https://dev.to/ag-grid/react-18-avoiding-use-effect-getting-called-twice-4i9e, I agree with the comment by `Alex Chiu`, is not a good practice.
'I wonder if it is possible to not do this in a useEffect?' --- I am not sure how to achieve this. I've tried several methods to resolve this issue, but none have been successful so far. Do you have a clear thought yet?

Reply:
Hi Starla,
I do not unfortunately :(. I think we just have to live with it? It's not worst solution imo, but at the minimum, strict mode is useable for people.

Reply:
Hi Andy,
Yes, I agree with living with it for now untill we could find a better solution to hide the warnnings.

Reply:
Hi Starla,
Just to clarify, this PR is good to go?

Reply:
Yes, I think so if we agree to live with the warnnings now.